linera-version/src/version_info.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    io::{Read as _, Write as _},
+    path::PathBuf,
+};
+
+#[cfg_attr(
+    linera_version_building,
+    derive(async_graphql::SimpleObject, serde::Deserialize, serde::Serialize)
+)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct CrateVersion {
+    pub major: u64,
+    pub minor: u64,
+    pub patch: u64,
+}
+
+impl From<semver::Version> for CrateVersion {
```

Comment:
Are we doing anything with `semver`? (Can't find it)

Reply:
It's returned by `cargo_metadata`, so we use it here just to convert out.

(I'd use `semver::Version` directly except that it isn't `async_graphql::Output`, et cetera — and making a remote type would require restating the structure anyway)

Reply:
(and, more frustratingly, also not `ToTokens` or constructible with a `const` function)